### PR TITLE
fix(gmp): Bug where onEvent could be null

### PR DESCRIPTION
**Pull request by staratlasjim**
## Changes

- Working through a bug where the GmEventService doesn't correctly handle the `onEvent` callback handler being null.
- If the health check service determines that the GmEventService needs to be restarted, the `onEvent` handler would be set to null.
- GmOrderBookService.ts: This fixes both the condition that inadvertently set the GmEventService::onEvent to null (line 116).
- Added logic to handle not having an `onEvent` handler in the GmEventService.

- some formatting changes due to prettier/eslint configs in my ide.

## Checklist

- [ ] UAT
- [ ] Passes final QA checks

### CODE REVIEW

**Comment by bravetarget** [comment]
Task linked: [CU-2nvdmn8 [Bug] Orders not loading in category orders view](https://app.clickup.com/t/2nvdmn8)

**Review by MikeLuDev — CHANGES_REQUESTED** [review]
Could you revert the changes to quotation marks?

File: src/marketplace/services/GmEventService.ts [hunk 1]
```diff
@@ -150,29 +166,55 @@ export class GmEventService {
...
+   * @param slotContext
+   * @protected
+   */
+  protected processEvent(
+    eventType: GmEventType,
+    order: Order,
+    slotContext: number
+  ): void {
+    // this can happen if StarCOMM has de-commissioned all rooms, or hasn't
+    // had a connection from a client and hasn't created rooms yet
```
Comment (MikeLuDev) [inline src/marketplace/services/GmEventService.ts:205]:
```suggestion
```

I'd leave this out - this is a public repository and the general audience doesn't have context on `StarComm` *yet*

Comment (staratlasjim) [reply src/marketplace/services/GmEventService.ts:205]: @mikeludev - removed those comments.  Also now that Prettier is in, I left the single quotes.

File: src/marketplace/services/GmOrderbookService.ts [hunk 2]
```diff
@@ -111,6 +113,8 @@ export class GmOrderbookService {
       this.marketplaceProgramId
     );
```
Comment (Buzzec) [inline src/marketplace/services/GmOrderbookService.ts:115]: Does `onEvent` need to be set to null here so that events are queued while re-initializing?

Comment (staratlasjim) [reply src/marketplace/services/GmOrderbookService.ts:115]: Great question: @Buzzec.  

Yes, it does, however it is already set to `null`

On line 109, it calls
-  `await this.gmEventService.end();`
- `GmEventService::End()`, line 112 sets `onEvent = null`.

As soon as `end` is called, the handler is null. 

Also, line 116 should remove the major use case where the event queue was the safety net, as `resetEventService` used to call `end` but not re-add the event handler.

**Review by MikeLuDev — APPROVED** [review]

**Pull request merged**